pywbemtools/pywbemcli/_common.py

```diff
@@ -456,6 +456,51 @@ def create_cimproperty(cim_type, is_array, name, value_str):
     return CIMProperty(name, cim_value, cim_type)
 
 
+def create_ciminstancename(cim_class, kv_keys):
+    """
+    Create a cim instance from the input parameters.
+
+      Parameters:
+
+        cim_class_: (:Class:`pywbem.CIMClass`)
+            The class from which the CIMInstance is to be created
+
+        properties ():
+            A tuple of name/value pairs representing the properties and their
+            values that are to be constructed for the instance. Required
+
+      Returns: CIMInstanceName with namespace = None and host = None
+
+      Exceptions:
+        click.ClickException if Property name not found in class or if mismatch
+          of property type in class vs value component of kv pair
+    """
+    # TODO make usable if no class can be acquired because GetClass not
+    # implemented in server.
```

Comment:
This TODO is unrelated to the code in this function. The CIMCLass object is already passed in, so if there is an issue with GetClass not being implemented, it would be wherever that GetClass is executed, and that's where this TODO should be.

Reply:
Moved to the common funciton in _cmd_instance.py and extended the idea somewhat. Still a todo and marked future

**DONE**